tsl/src/chunk_api.c

```diff
@@ -371,3 +372,126 @@ chunk_create(PG_FUNCTION_ARGS)
 
 	PG_RETURN_DATUM(HeapTupleGetDatum(tuple));
 }
+
+
+/*
+ * Detach a chunk from a hypertable.
+ */
+Datum
+chunk_detach(PG_FUNCTION_ARGS)
+{
+	Oid chunk_relid = PG_ARGISNULL(0) ? InvalidOid : PG_GETARG_OID(0);
+	Cache *hcache;
+	Hypertable *ht;
+	Chunk *chunk;
+
+	TS_PREVENT_FUNC_IF_READ_ONLY();
+
+	chunk = ts_chunk_get_by_relid(chunk_relid, true);
+	if (!chunk)
+		ereport(ERROR,
+				(errcode(ERRCODE_UNDEFINED_TABLE),
+				 errmsg("chunk with OID %u does not exist", chunk_relid)));
```

Comment:
You're passing in `true` to the function above, so it should not return NULL, hence this error message can never show. (And it is using `ERRCODE_UNDEFINED_TABLE` while `ts_chunk_get_by_relid` will use `ERRCODE_UNDEFINED_OBJECT`, so it's that too.)

As `Assert` (or `Ensure`) might be prudent though, just to capture bugs.